src/php/Api/Domain/PhelFnNormalizer.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Api\Domain;
+
+use Phel\Api\Infrastructure\PhelFnLoaderInterface;
+use Phel\Api\Transfer\NormalizedPhelFunction;
+use Phel\Lang\Keyword;
+
+final class PhelFnNormalizer implements PhelFnNormalizerInterface
+{
+    public function __construct(
+        private PhelFnLoaderInterface $phelFnLoader,
+    ) {
+    }
+
+    /**
+     * @return array<string,list<NormalizedPhelFunction>>
+     */
+    public function getNormalizedGroupedFunctions(): array
+    {
+        $normalizedData = $this->phelFnLoader->getNormalizedPhelFunctions();
+
+        $result = [];
+        foreach ($normalizedData as $fnName => $meta) {
+            $isPrivate = $meta[Keyword::create('private')] ?? false;
+            if ($isPrivate) {
+                continue;
+            }
+
+            $doc = $meta[Keyword::create('doc')] ?? '';
+            $pattern = '#(```phel\n(?<fnSignature>.*)\n```\n)?(?<desc>.*)#s';
+            preg_match($pattern, $doc, $matches);
+
+            $result[$this->groupKey($fnName)][] = new NormalizedPhelFunction(
+                $fnName,
+                $doc,
+                $matches['fnSignature'] ?? '',
+                $this->formatDescription($matches['desc'] ?? ''),
+            );
+        }
+
+        foreach ($result as $values) {
+            usort($values, $this->sortingPhelFunctionsCallback());
+        }
+
+        return $result;
+    }
+
+    private function groupKey(string $fnName): string
+    {
+        $key = preg_replace(
+            '/[^a-zA-Z0-9\-]+/',
+            '',
+            str_replace('/', '-', $fnName),
+        );
+
+        return strtolower(rtrim($key, '-'));
+    }
+
+    /**
+     * The $desc is in Markdown format, the regex transforms links `[printf](https://...)` into `<i>printf</i>`.
+     */
+    private function formatDescription(string $desc): string
+    {
+        return preg_replace(
```

Comment:
We are doing this `markdown` to `HTML` translation here (in the `Domain` layer!), do we want to keep it here or maybe do this transformation somewhere else at a lower level (maybe defining the output in some extra params in the constructor of this class and/or even in `phel-lang.org` project while we output the data?)

Reply:
Yes, I will change it to keep the link as `<a>` instead of `<i>`. I think it's ok to keep this thing in the domain layer.